symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/StreamService.java

```diff
@@ -75,23 +84,59 @@ public OboStreamService obo(AuthSession oboSession) {
   /**
    * {@inheritDoc}
    */
-  public V2StreamAttributes getStreamInfo(String streamId) {
+  @Override
+  public V2StreamAttributes getStreamInfo(@Nonnull String streamId) {
     return executeAndRetry("getStreamInfo",
         () -> streamsApi.v2StreamsSidInfoGet(streamId, authSession.getSessionToken()));
   }
 
   /**
    * {@inheritDoc}
    */
-  public List<StreamAttributes> listStreams(StreamFilter filter) {
+  @Override
+  public List<StreamAttributes> listStreams(@Nullable StreamFilter filter) {
     return executeAndRetry("listStreams",
         () -> streamsApi.v1StreamsListPost(authSession.getSessionToken(), null, null, filter));
   }
 
   /**
    * {@inheritDoc}
    */
-  public void addMemberToRoom( Long userId, String roomId) {
+  @Override
+  public List<StreamAttributes> listStreams(@Nullable StreamFilter filter, @Nullable PaginationAttribute pagination) {
+    if (pagination != null) {
+      return executeAndRetry("listStreams",
+          () -> streamsApi.v1StreamsListPost(authSession.getSessionToken(), pagination.skip(), pagination.limit(), filter));
+    }
+    return listStreams(filter);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public java.util.stream.Stream<StreamAttributes> listStreamsInStream(@Nullable StreamFilter filter) {
+    return listStreamsInStream(filter, null);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public java.util.stream.Stream<StreamAttributes> listStreamsInStream(@Nullable StreamFilter filter,
+      @Nullable StreamPaginationAttribute pagination) {
+    PaginatedApi<StreamAttributes> api = (offset, limit) -> listStreams(filter, new PaginationAttribute(offset, limit));
+
+    pagination = PaginatedService.getValidPaginationAttribute(pagination, 50, 50);
```

Comment:
it is usually considered a bad practice to reassign parameters, we can instead a local variable with a different/more meaningful name

Reply:
can't we prevent the creation of an invalid pagination attribute instead? (by throwing  illegal argument exception in the constructor)

Reply:
i.e I'd rather enforce the creation of valid objects rather than silently fixing the objects for the user